golem-worker-service-base/src/gateway_execution/request.rs

```diff
@@ -29,13 +29,25 @@ const COOKIE_HEADER_NAMES: [&str; 2] = ["cookie", "Cookie"];
 /// Thin wrapper around a poem::Request that is used to evaluate all binding types when coming from an http gateway.
 pub struct RichRequest {
     pub underlying: poem::Request,
```

Comment:
I tried this private too, but too many compilation errors. Not the purpose of this PR